x-pack/plugins/task_manager/server/task_store.ts

```diff
@@ -295,28 +291,18 @@ export class TaskStore {
     claimTasksById: OwnershipClaimingOpts['claimTasksById'],
     size: OwnershipClaimingOpts['size']
   ): Promise<ConcreteTaskInstance[]> {
+    const claimedTasksQuery = tasksClaimedByOwner(this.taskManagerId);
     const { docs } = await this.search({
-      query: {
-        bool: {
-          must: [
-            {
-              term: {
-                'task.ownerId': this.taskManagerId,
-              },
-            },
-            { term: { 'task.status': 'claiming' } },
-          ],
-        },
-      },
-      size,
-      sort:
+      query:
         claimTasksById && claimTasksById.length
-          ? sortByIdsThenByScheduling(claimTasksById)
-          : SortByRunAtAndRetryAt,
+          ? asPinnedQuery(claimTasksById, claimedTasksQuery)
+          : claimedTasksQuery,
+      size,
+      sort: SortByRunAtAndRetryAt,
       seq_no_primary_term: true,
     });
 
-    return docs;
+    return claimTasksById && claimTasksById.length ? docs : docs;
```

Comment:
This is returning `docs` regardless of the value of the predicate.  Did we need to filter the docs somehow returned when using the pinned query?  This would be a place to do it ...

Reply:
Left overs that hadn't been cleaned up yet :)